ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -2186,6 +2195,43 @@ private TaskRunner launchTask(Task<? extends Serializable> tsk, String queryId,
     return tskRun;
   }
 
+  private void enforceQueue(HiveConf conf) {
+    String queueEnforcerClassName = conf.getVar(HiveConf.ConfVars.QUEUE_ENFORCER_CLASS);
+    QueueEnforcer queueEnforcer = null;
+    try {
+      queueEnforcer = (QueueEnforcer) Class.forName(queueEnforcerClassName)
+          .newInstance();
+    } catch (Throwable e) {
+      throw new RuntimeException("Couldn't enforce queue due to " + e.getMessage());
+    }
+    String enforcedQueue = queueEnforcer
+        .getEnforcedQueue(conf.get(Constants.MAPRED_QUEUE_NAME),
+            getInitiator(conf), conf);
+    conf.set(Constants.MAPRED_QUEUE_NAME, enforcedQueue);
+    conf.set(Constants.TEZ_QUEUE_NAME, enforcedQueue);
+  }
+
+  private String getInitiator(HiveConf conf) {
+    if (org.apache.parquet.Strings.isNullOrEmpty(conf.get(Constants.INITIATOR))) {
+      return UserNameWrapper.INSTANCE.getUsername();
+    }
+    return conf.get(Constants.INITIATOR);
+  }
+
+  private void enrichJobName(Task<? extends Serializable> tsk, String queryId, int jobs) {
+    JobNameEnricher jobEnricher = null;
+    try {
+      jobEnricher = (JobNameEnricher) Class.forName(conf.getVar(ConfVars.JOBNAME_ENRICHER_CLASS))
```

Comment:
Same. Lets have this initialised elsewhere and reuse. Lets discuss where we can do this.